src/main/java/arpa/home/nustudy/utils/Storage.java

```diff
@@ -84,4 +101,81 @@ public void save(CourseManager courses, SessionManager sessions) {
             logger.log(Level.SEVERE, "Unable to save courses: ");
         }
     }
+
+    /**
+     * Parses then loads the stored data from a buffered reader into the {@code CourseManager} and {@code
+     * SessionManager} instances.
+     *
+     * @param courses The {@code CourseManager} instance to load in the loaded courses.
+     * @param sessions The {@code SessionManager} instance to load in the loaded sessions.
+     * @param buffer The duplicate copy of type {@code BufferedReader} of the storage file.
+     * @throws IOException If an I/O error is raised while reading from the buffer.
+     */
+    private static void loadData(CourseManager courses, SessionManager sessions,
+            final BufferedReader buffer) throws IOException {
+        String line;
+        while ((line = buffer.readLine()) != null) {
+            line = line.trim();
+
+            try {
+                if (line.startsWith("C|")) {
+                    final Course course = Course.fromStorageString(line);
+                    courses.add(course);
+                }
+                if (line.startsWith("S|")) {
+                    final Object[] courseAndSession = Session.fromStorageString(line);
+                    final String courseName = (String) courseAndSession[0];
+                    final Integer sessionHours = (Integer) courseAndSession[1];
+
+                    Course matchingCourse = null;
+                    for (Course c : courses) {
```

Comment:
Nit: add newline in between


```suggestion
                    Course matchingCourse = null;

                    for (Course c : courses) {
```

Reply:
Noted thanks.